nucliadb/nucliadb/ingest/orm/entities.py

```diff
@@ -140,14 +140,19 @@ async def set_entities_group_force(self, group: str, entitiesgroup: EntitiesGrou
         await self.index_entities_group(group, entitiesgroup)
 
     async def delete_entities(self, group: str, delete: List[str]):
-        entities_group = await self.get_stored_entities_group(group)
-        if entities_group is None:
+        stored = await self.get_stored_entities_group(group)
+        indexed = await self.get_indexed_entities_group(group)
+
+        if stored is None and indexed is None:
             return
+
+        stored = stored or EntitiesGroup()
+        indexed = indexed or EntitiesGroup()
         for name in delete:
-            if name in entities_group.entities:
-                entity = entities_group.entities[name]
+            if name in stored.entities or name in indexed.entities:
+                entity = stored.entities[name]
```

Comment:
this could raise a KeyError if `name` is in `indexed` but not in `stored`?

Reply:
I don't think so, as they are protobufs, `stored.entities[name]` returns an empty `Entity` if the key doesn't exist. The conditional won't fail either.